content/articles/dockerized-prisma-postgres-api/index.md

```diff
@@ -0,0 +1,516 @@
+At some point, you will need a database for your back-end application, be it with Python, Java, Node.js, etc. This is where a connection to the database is needed. For the backed to communicate with the database of your choice, you need a way to connect your server to that database. So when someone sends a request to the server, the server modifies that data and writes it to the database. When someone writes a query requesting data, the server reads that query and sends it to the database then send a repose to the user. This is opposed to just reading from one of our static arrays.
+
+Now to connect to that database, you can use native drivers. Native divers are mainly libraries that allow you to connect to a database. For example, when working with Node.js and a MySQL database, you can choose to use a MySQL native drive provided by the Node.js NPM. However, they do not provide a way to modify or validate the data structures or a way to model the relationships between the databases. This means you can't model your data relations, seed, or migrate it. With native drivers, the database connection is natively implemented using the barebones of the programming language being used.
+
+Other options would be to use ORM (object-relational mapping). For example, when working with Node.js, MongoDB provide a mongoose (an NPM library) database driver. This makes it easy for a developer to connect a Node.js back-end to a MongoDB database. ORM will introduce you to some new features. This includes modeling, validating, migrating data, describing the relationships between different data fields, etc. For example, if you are using an SQL database, you can model your schema and the fields you want to use. You can validate your data model, for example, by saying a password field needs to have at least six characters, a certain field needs to be integers, etc. Manipulating this data model becomes easier as you can add and remove fields as the data structure changes over time.
+
+Prisma falls under the category of an ORM, and it has all features of a typical ORM, such as mongoose and sequelize. Unlike the other ORM, Prisma comes with a tone of other added features. Prisma is database agnostic. Mongoose only works on a MongoDB database, and sequelize only works on SQL-based databases. If you decide to switch from MongoDB to PostgreSQL, you have to rewrite most of your applications since the two libraries are so different. You don't have to do that with Prisma. Prisma [supports major database](https://www.prisma.io/docs/reference/database-reference/supported-databases) such as MySQL, PostgreSQL, MongoDB, MariaDB, Azure SQL, Microsoft SQL Server AWS Aurora, and AWS Aurora Serverless. This means that you can easily pick the database that fits the code structure of your application and even switch between databases without the need to change any code structure of your application.
```

Comment:
Can you break down these paragraphs into smaller ones? One paragraph must talk about one idea.

Also, add necessary reference links. For example, for ORM, MongoDB, AWS Aurora and so on.